### ISSUE TITLE
False positive S3900

### ISSUE THREAD

**Issue by jasper-d** [SonarSource/sonar-dotnet#2476]
### Description

I observed a false positive for [rule 3900](https://rules.sonarsource.com/csharp/RSPEC-3900). 

### Repro steps
Consider the following method:

```
public void Foo(params string[] infixes)
{
    if(infixes == null  || infixes.Length == 0) {
        infixes = new string[1];
    }
    else
    {
        Array.Resize(ref infixes, infixes.Length + 1); // S3900, Refactor this method to add validation of parameter 'infixes' before using it.
    }
    // more stuff
}

```

Run it through Sonar.

### Expected behavior

No warning, I dont see how infixes could be null in the else branch.

### Actual behavior

S3900 violation is reported by Sonar

### Known workarounds

-

### Related information

* SonarC# Version: 7.15 build 8572
* Visual Studio Version: 16.1.5
* Scanner for MSBuild Version: 4.6.2
